### PR TITLE
[Merged by Bors] - Use the new Poet Submit API

### LINKED ISSUE #3708: use the new extended Poet Submit API

### PR DESCRIPTION
**Pull request by poszu**
## Motivation
Part of #3458 
Closes #3708

## Changes
Updated the poet client and the nipost builder to use the new extended Poet Submit API.

## Test Plan
- unit tests
- e2e system tests with the new Poet

## DevOps Notes
- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by poszu** [comment]
bors try

File: activation/activation_test.go [hunk 1]
```diff
@@ -105,38 +105,39 @@ func NewMockSigner() *MockSigning {
...
 
-func (np *NIPostBuilderMock) BuildNIPost(_ context.Context, challenge *types.Hash32, commitmentAtx types.ATXID, _ time.Time) (*types.NIPost, time.Duration, error) {
+func (np *NIPostBuilderMock) BuildNIPost(_ context.Context, challenge *types.PoetChallenge, commitmentAtx types.ATXID, _ time.Time) (*types.NIPost, time.Duration, error) {
 	if np.buildNIPostFunc != nil {
 		return np.buildNIPostFunc(challenge, commitmentAtx)
 	}
-	return newNIPostWithChallenge(challenge, np.poetRef), 0, nil
+	hash, err := challenge.Hash()
+	if err != nil {
+		panic(fmt.Sprintf("failed to hash challenge (%v)", err))
```
Comment (countvonzero) [inline activation/activation_test.go]: please do log.fatal here to avoid useless stack trace here.

File: activation/nipost.go [hunk 2]
```diff
@@ -109,16 +119,34 @@ func (nb *NIPostBuilder) BuildNIPost(ctx context.Context, challenge *types.Hash3
...
+		signature := nb.signer.Sign(challenge)
+		poetRequests := nb.submitPoetChallenges(ctx, challenge, signature)
 		if ctx.Err() != nil {
 			return nil, 0, ctx.Err()
 		}
-		if len(poetRequests) == 0 {
+
+		validPoetRequests := make([]types.PoetRequest, 0, len(poetRequests))
+		for _, req := range poetRequests {
+			if !bytes.Equal(req.PoetRound.ChallengeHash, challengeHash[:]) {
```
Comment (countvonzero) [inline activation/nipost.go:134]: curious: is this a result of malicious poet server?

**Review by countvonzero — APPROVED** [review]

File: activation/activation_test.go (hunk 1 above)
Comment (poszu) [reply activation/activation_test.go]: Good idea, thanks :)

File: activation/nipost.go (hunk 2 above)
Comment (poszu) [reply activation/nipost.go:134]: Malicious/buggy poet service or its trusted gateway (which actually hashes the challenge). It's a sanity check - if the poet was unable to properly hash the challenge then there is no chance that the challenge will be included in its proof.

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
Bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -314,8 +314,9 @@ jobs:
           node_selector: cloud.google.com/gke-nodepool=gha
           size: 50
           bootstrap: 10m
-          level: info
+          level: debug
           clusters: 2
+          poet_image: us-east4-docker.pkg.dev/spacemesh-infra-core/docker-dev/spacemeshos/poet:poet-challenge-with-verifying-svc
```
Comment (poszu) [inline .github/workflows/ci.yml]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -327,8 +328,9 @@ jobs:
           node_selector: cloud.google.com/gke-nodepool=gha
           size: 30
           bootstrap: 10m
-          level: info
+          level: debug
           clusters: 2
+          poet_image: us-east4-docker.pkg.dev/spacemesh-infra-core/docker-dev/spacemeshos/poet:poet-challenge-with-verifying-svc
```
Comment (poszu) [inline .github/workflows/ci.yml]:
```suggestion
```

**Comment by poszu** [comment]
bors merge

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors try

**Comment by poszu** [comment]
bors merge

**Pull request closed without merging**